mz_zip.c

```diff
@@ -787,12 +788,16 @@ static int32_t mz_zip_entry_write_header(void *stream, uint8_t local, mz_zip_fil
     }
 
     if (err == MZ_OK) {
-        if (mz_stream_write(stream, filename, filename_length) != filename_length)
+        char conformed_filename[512];
```

Comment:
It would be better to just use `MZ_ALLOC` here, and if it returns NULL then return `MZ_ERROR_MEM`.

Reply:
Thanks @nmoinvaz, modifications checked in.